internal/engine/handler.go

```diff
@@ -95,11 +124,15 @@ func (e *ExecutorEventHandler) HandleEntityEvent(msg *message.Message) error {
 		if inf.Type == pb.Entity_ENTITY_ARTIFACTS {
 			time.Sleep(ArtifactSignatureWaitPeriod)
 		}
-		// TODO: Make this timeout configurable
-		ctx, cancel := context.WithTimeout(e.terminationcontext, DefaultExecutionTimeout)
+
+		ctx, cancel := context.WithTimeout(msgCtx, DefaultExecutionTimeout)
 		defer cancel()
+		ctx = engcontext.WithEntityContext(ctx, &engcontext.EntityContext{
+			Project: engcontext.Project{ID: inf.ProjectID},
+			// TODO: extract Provider name from ProviderID
```

Comment:
IMHO, if we do decide to retain the provider as a feature flag attribute, we should consider using the provider ID instead of the name.

Reply:
Alternatively, we should use the provider _class_, rather than a particular provider ID.  (i.e. "enable this feature for users with the foo provider")  It still feels like a stretch, though.

I think I'll just remove this.